app/services/teacher_training_public_api/sync_sites.rb

```diff
@@ -112,5 +113,16 @@ def handle_course_options_with_invalid_sites(sites)
         course_option.update!(site_still_valid: false)
       end
     end
+
+    def handle_course_options_with_reinstated_sites(sites)
+      withdrawn_course_options = @course.course_options.joins(:site).where(site_still_valid: false)
+      site_codes = sites.map(&:code)
```

Comment:
Site codes aren't unique enough to just check those, we also need to take into account the provider (we have an index on sites so that they are unique on code and provider_id)

Reply:
We validate that the `course` and `site` each `CourseOption` links are associated with the same `Provider` so the sites returned here should be scoped to a single provider already.

Reply:
Ah yes, I missed this, sorry!